File: Demo Projects/APIcreationLearning/app/src/main/java/com/example/apicreationlearning/CustomAdapter.java
package com.example.apicreationlearning;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;


public class CustomAdapter extends RecyclerView.Adapter<CustomAdapter.CustomViewHolder>
{
    private List<PostPOJO> dataList;
    private Context context;
    public CustomAdapter(Context context, List<PostPOJO> dataList)
    {this.context=context;
    this.dataList=dataList;


    }
    class CustomViewHolder extends RecyclerView.ViewHolder
    {
        TextView title,posts;

        CustomViewHolder(View itemView)
        {
            super(itemView);
            title=itemView.findViewById(R.id.title);
            posts=itemView.findViewById(R.id.body);

        }
    }
    @Override
    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
    {
        LayoutInflater layoutInflater=LayoutInflater.from(parent.getContext());
        View view =layoutInflater.inflate(R.layout.design,parent,false);
        return  new CustomViewHolder(view);
    }

    @Override
    public void onBindViewHolder(CustomViewHolder holder,int position)
    {
        holder.title.setText(dataList.get(position).getTitle());
        holder.posts.setText(dataList.get(position).getBody());
    }

    @Override
    public int getItemCount()
    {
        return dataList.size();
    }







}

File: Latitude_projects/Bottom_navigation/app/src/main/java/com/example/bottom_navigation/Adapter/CustomAdapter.kt
package com.example.bottom_navigation.Adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.bottom_navigation.ItemsViewModel
import com.example.bottom_navigation.R

class CustomAdapter(private val mList: MutableList<ItemsViewModel>) :
    RecyclerView.Adapter<CustomAdapter.ViewHolder>() {

    fun removeItem(position: Int) {
        mList.removeAt(position)
        notifyItemRemoved(position)
        notifyItemRangeChanged(position, mList.size)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        // inflates the card_view_design view
        // that is used to hold list item
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.card_view_design, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val itemViewModel = mList[position]

        // sets the image to the imageview from our itemHolder class
        holder.imageView.setImageResource(itemViewModel.image)

        // sets the text to the textview from our itemHolder class
        holder.textView.text = itemViewModel.text




        // Set click listener for the button in each item
        holder.btn.setOnClickListener {
            // Implement the action you want to perform when the button is clicked
            // For example, you can remove the item from the list and update the RecyclerView
            removeItem(position)
        }
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imageView: ImageView = itemView.findViewById(R.id.imageview)
        val textView: TextView = itemView.findViewById(R.id.textView)
        val btn: Button = itemView.findViewById(R.id.btnDelete)
    }
}
